### PR TITLE
Refactor index command line program

### PR DESCRIPTION
**Pull request by rjgildea**
- Simple function instead of class
- Use sys.exit instead of Sorry

### CODE REVIEW

File: command_line/index.py [hunk 1]
```diff
@@ -119,106 +122,81 @@ def index_experiments(experiments, reflections, params, known_crystal_models=Non
...
+    reflections = reflections[0]
+
+    for expt in experiments:
+        if (
+            expt.goniometer is not None
+            and expt.scan is not None
+            and expt.scan.get_oscillation()[1] == 0
+        ):
+            expt.goniometer = None
+            expt.scan = None
```
Comment (benjaminhwilliams) [inline command_line/index.py]: Can we have a comment above that describes in plain English the case that this is designed to catch?

File: command_line/index.py [hunk 2]
```diff
@@ -119,106 +122,81 @@ def index_experiments(experiments, reflections, params, known_crystal_models=Non
...
+        known_crystal_models = experiments.crystals()
+    else:
+        known_crystal_models = None
+
+    if len(reflections) == 0:
+        raise ValueError("No reflection lists found in input")
+    elif len(reflections) == 1:
+        reflections[0]["imageset_id"] = reflections[0]["id"]
+    elif len(reflections) > 1:
+        assert len(reflections) == len(experiments)
```
Comment (benjaminhwilliams) [inline command_line/index.py]: Is it not also necessary to make this assertion if `len(reflections) == 1`?

File: command_line/index.py [hunk 3]
```diff
@@ -119,106 +122,81 @@ def index_experiments(experiments, reflections, params, known_crystal_models=Non
...
-                    refl["imageset_id"] = flex.size_t(len(refl), 0)
-                    futures.append(
-                        pool.submit(
-                            index_experiments,
-                            ExperimentList([expt]),
-                            refl,
-                            copy.deepcopy(params),
-                            known_crystal_models=known_crystal_models,
-                        )
+def index(experiments, reflections, params):
```
Comment (benjaminhwilliams) [inline command_line/index.py]: DIALS and xia2 are riddled with functions called `index`, `indexer`, etc..  Here, it's definitely appropriate, but to make it clear what _this_ function does (it's the main one after all), I think it would be quite handy to have a docstring.

File: command_line/index.py (hunk 2 above)
Comment (rjgildea) [reply command_line/index.py]: No, because in this case the input could multiple experiments, but with the reflections corresponding to those experiments concatenated into a single reflection table (rather than a list of reflection tables one per experiment).

Comment (benjaminhwilliams) [reply command_line/index.py]: Ah, OK, thanks.

File: command_line/index.py (hunk 1 above)
Comment (benjaminhwilliams) [reply command_line/index.py]: @dagewa, I see you've been digging around here today.  Any chance you could distil your new wisdom into an explanatory comment?
